### PR TITLE
Changed "Eval" call to use "check" method.

### PR DESCRIPTION
**Pull request by jas-valgotar**
Changed "Eval" call to use "check" method, so that we can leverage "CheckResult".

### CODE REVIEW

**Review by MikeStall — DISMISSED** [review]
:shipit:

File: Samples/WebDemo/Controllers/EvaluationController.cs [hunk 1]
```diff
@@ -42,7 +42,13 @@ public IActionResult HandleRequest([FromBody] RequestBody body)
...
                 
                 // TODO - this should be eval Async and catch timeouts. 
-                var result = engine.Eval(body.expression, parameters);
+                if (parameters == null)
+                {
+                    parameters = RecordValue.Empty();
+                }
+                var check = engine.Check(body.expression, parameters.Type, options:null);
+                check.ThrowOnErrors();
+                var result =  check.Expression.EvalAsync(parameters, cancel: CancellationToken.None).Result;
```
Comment (MikeStall) [inline Samples/WebDemo/Controllers/EvaluationController.cs:51]: >.Result [](http://example.com/codeflow?start=100&length=7)

We could actually get rid of this, and make the method be 

async Task<IActionResult>

and then await this. 

Then we could also add cancellation support, have it just abort after 5 seconds. 
See test case here:
[https://github.com/microsoft/Power-Fx/blob/25e9106435fc6c4fa2e8208bea192d8a02dfcead/src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineAsyncTests.cs#L173](https://github.com/microsoft/Power-Fx/blob/25e9106435fc6c4fa2e8208bea192d8a02dfcead/src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineAsyncTests.cs#L173)

Comment (jas-valgotar) [reply Samples/WebDemo/Controllers/EvaluationController.cs]: As Discussed, added abort after 2 seconds.

File: Samples/WebDemo/Controllers/EvaluationController.cs [hunk 2]
```diff
@@ -52,6 +60,21 @@ public IActionResult HandleRequest([FromBody] RequestBody body)
...
+                cts.Dispose();
+            }
+        }
+
+        private async Task<FormulaValue> EvalAsync(IPowerFxEngine engine, string expressionText, CancellationToken cancel, RecordValue parameters = null, ParserOptions options = null)
+        {
+            if (parameters == null)
+            {
+                parameters = RecordValue.Empty();
+            }
```
Comment (MikeStall) [inline Samples/WebDemo/Controllers/EvaluationController.cs:74]: repeated with line 49?

File: Samples/WebDemo/Controllers/EvaluationController.cs [hunk 3]
```diff
@@ -1,49 +1,57 @@
...
 {
-    // Called by the demo page to evaluate an expression. 
+    // Called by the demo page to evaluate an expression.
     [ApiController]
     [Route("eval")]
     public class EvaluationController : ControllerBase
     {
         private readonly ILogger<EvaluationController> _logger;
-        
+        public static TimeSpan timeout = TimeSpan.FromSeconds(2);
```
Comment (MikeStall) [inline Samples/WebDemo/Controllers/EvaluationController.cs:21]: >timeout [](http://example.com/codeflow?start=31&length=7)

coding conventions comment - fields should start with a '_'.

Comment (jas-valgotar) [reply Samples/WebDemo/Controllers/EvaluationController.cs]: done!

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**